release/pkg/generate_spec.go

```diff
@@ -392,76 +395,85 @@ func (r *ReleaseConfig) GetSourceImageURI(name, repoName string, tagOptions map[
 
 func (r *ReleaseConfig) GetReleaseImageURI(name, repoName string, tagOptions map[string]string) string {
 	var releaseImageUri string
-	var semVer string
-	if r.DevRelease {
-		semVer = r.DevReleaseUriVersion
-	} else {
-		semVer = fmt.Sprintf("%d", r.BundleNumber)
-	}
 
 	if name == "bottlerocket-bootstrap" {
-		releaseImageUri = fmt.Sprintf("%s/%s:v%s-%s-eks-a-%s",
+		releaseImageUri = fmt.Sprintf("%s/%s:v%s-%s-eks-a",
 			r.ReleaseContainerRegistry,
 			repoName,
 			tagOptions["eksDReleaseChannel"],
 			tagOptions["eksDReleaseNumber"],
-			semVer,
 		)
 	} else if name == "cloud-provider-vsphere" {
-		releaseImageUri = fmt.Sprintf("%s/%s:%s-eks-d-%s-eks-a-%s",
+		releaseImageUri = fmt.Sprintf("%s/%s:%s-eks-d-%s-eks-a",
 			r.ReleaseContainerRegistry,
 			repoName,
 			tagOptions["gitTag"],
 			tagOptions["eksDReleaseChannel"],
-			semVer,
 		)
 	} else if name == "eks-anywhere-cluster-controller" {
 		if r.DevRelease {
-			releaseImageUri = fmt.Sprintf("%s/%s:v0.0.0-eks-a-%s",
+			releaseImageUri = fmt.Sprintf("%s/%s:v0.0.0-eks-a",
 				r.ReleaseContainerRegistry,
 				repoName,
-				semVer,
 			)
 		} else {
-			releaseImageUri = fmt.Sprintf("%s/%s:%s-eks-a-%s",
+			releaseImageUri = fmt.Sprintf("%s/%s:%s-eks-a",
 				r.ReleaseContainerRegistry,
 				repoName,
 				r.ReleaseVersion,
-				semVer,
 			)
 		}
 	} else if name == "eks-anywhere-diagnostic-collector" {
 		if r.DevRelease {
-			releaseImageUri = fmt.Sprintf("%s/%s:v0.0.0-eks-a-%s",
+			releaseImageUri = fmt.Sprintf("%s/%s:v0.0.0-eks-a",
 				r.ReleaseContainerRegistry,
 				repoName,
-				semVer,
 			)
 		} else {
-			releaseImageUri = fmt.Sprintf("%s/%s:%s-eks-a-%s",
+			releaseImageUri = fmt.Sprintf("%s/%s:%s-eks-a",
 				r.ReleaseContainerRegistry,
 				repoName,
 				r.ReleaseVersion,
-				semVer,
 			)
 		}
 	} else if name == "kind-node" {
-		releaseImageUri = fmt.Sprintf("%s/%s:%s-eks-d-%s-%s-eks-a-%s",
+		releaseImageUri = fmt.Sprintf("%s/%s:%s-eks-d-%s-%s-eks-a",
 			r.ReleaseContainerRegistry,
 			repoName,
 			tagOptions["kubeVersion"],
 			tagOptions["eksDReleaseChannel"],
 			tagOptions["eksDReleaseNumber"],
-			semVer,
 		)
 	} else {
-		releaseImageUri = fmt.Sprintf("%s/%s:%s-eks-a-%s",
+		releaseImageUri = fmt.Sprintf("%s/%s:%s-eks-a",
 			r.ReleaseContainerRegistry,
 			repoName,
 			tagOptions["gitTag"],
-			semVer,
 		)
 	}
 
-	return releaseImageUri
+	var semVer string
+	if r.DevRelease {
+		currentSourceImageUri := r.GetSourceImageURI(name, repoName, tagOptions)
+		previousReleaseImageUri := fmt.Sprintf("%s-%s", releaseImageUri, r.PreviousDevReleaseUriVersion)
+		sameDigest := r.CompareHashWithPreviousBundle(currentSourceImageUri, previousReleaseImageUri)
+		if sameDigest {
+			semVer = r.PreviousDevReleaseUriVersion
+		} else {
+			semVer = r.DevReleaseUriVersion
+		}
+	} else {
+		semVer = fmt.Sprintf("%d", r.BundleNumber)
```

Comment:
I think the idea is to get it working in dev first and then come back to staging/prod since they are different beast in the way they build/copy to public ecr?